core/src/usage/type_params.rs

```diff
@@ -226,11 +226,13 @@ impl UsesTypeParams for syn::GenericArgument {
     fn uses_type_params<'a>(&self, options: &Options, type_set: &'a IdentSet) -> IdentRefSet<'a> {
         match *self {
             syn::GenericArgument::Type(ref v) => v.uses_type_params(options, type_set),
-            syn::GenericArgument::Binding(ref v) => v.uses_type_params(options, type_set),
+            syn::GenericArgument::AssocType(ref v) => v.uses_type_params(options, type_set),
             syn::GenericArgument::Constraint(ref v) => v.uses_type_params(options, type_set),
-            syn::GenericArgument::Const(_) | syn::GenericArgument::Lifetime(_) => {
-                Default::default()
-            }
+            syn::GenericArgument::AssocConst(_)
+            | syn::GenericArgument::Const(_)
+            | syn::GenericArgument::Lifetime(_) => Default::default(),
+            // non-exhaustive enum
+            _ => Default::default(),
```

Comment:
Returning `default` here feels odd, since it's going to cause us to silently miss type params in new syntax constructs.

Reply:
Do you have a better suggestion how to handle non-exhaustive enums? The functions aren't fallible, so returning error is not an option.

Reply:
I was wondering if there's some use of `stringify` that would get the job done, but I'm not sure that's the case.

Reply:
What would `stringify` do here? The function needs to return a `IdentRefSet`, not a `String`. The default branch could return `Err`. However, none of the functions are currently fallible. All cases of `_ => Default::default(),` could be turned into an `Err` and the `Result`s propagated through the call stack. This will be fragile and future `syn` updates will introduce new variants and then `darling` will break and require code updates. This can also happen right now, but there is a chance that `Default::default` is actually the correct value. Thus, breakages would be less frequent.

I can change everything to be fallible, but I first would like to know if that is the correct approach.